### PR TITLE
[SYCL][Libdevice] Split SYCL specific code in sanitizer libdevice

### PR DESCRIPTION
**Pull request by jinge90**
Current SYCL device sanitizer can be ported to other offloading solution such as openmp but sanitizer libdevice uses some SYCL specific macro or code which is only valid in SYCL scenario, this PR splits these code/macros.

### CODE REVIEW

**Review by zhaomaosu — APPROVED** [review]
lgtm

**Comment by jinge90** [comment]
Hi, @intel/llvm-reviewers-runtime 
Could you help review this PR?
Thanks very much.

**Comment by jinge90** [comment]
Hi, @sergey-semenov 
Could you take a look at this PR?
Thanks very much.

File: libdevice/sanitizer_utils.cpp [hunk 1]
```diff
@@ -251,6 +257,12 @@ bool __asan_region_is_value(uptr addr, int32_t as, std::size_t size,
   return true;
 }
 
+#if __SYCL_DEVICE_ONLY__
+#define __DEVICE_SANITIZER_REPORT_ACCESSOR __DeviceSanitizerReportMem.get()
+#else
+#define __DEVICE_SANITIZER_REPORT_ACCESSOR
```
Comment (sergey-semenov) [inline libdevice/sanitizer_utils.cpp:263]: Could you please elaborate a bit on what the expected outcome in this `#else` branch is? Looks like the compilation will just fail in that case.

Comment (jinge90) [reply libdevice/sanitizer_utils.cpp:263]: Hi, @sergey-semenov 
Yes, the file must be built with __SYCL_DEVICE_ONLY__ or corresponding openmp macro defined, it not, we expect a compfail to be reported for notification. The openmp support will be added later.
Thanks very much

**Review by sergey-semenov — APPROVED** [review]

**Comment by jinge90** [comment]
Hi, @sergey-semenov 
Sorry for bothering again, I just updated the patch to replace "#if __SYCL_DEVICE_ONLY__" with "#ifdef ..." just to unify with previous code. Could you approve again?
Thanks very much.

**Review by sergey-semenov — APPROVED** [review]

**Comment by jinge90** [comment]
Hi, @intel/llvm-gatekeepers 
Could you help merge this patch?
Thanks very much.

**Pull request merged**